Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -88,11 +88,19 @@ async function doWork() {
 
         let downloadCommitsAndWorkItems: boolean = tl.getBoolInput("downloadCommitsAndWorkItems", false);
         if (downloadCommitsAndWorkItems) {
-            new ArtifactDetailsDownloader()
-                .DownloadCommitsAndWorkItems(jenkinsJobDetails)
-                .then(
-                () => console.log(tl.loc("SuccessfullyDownloadedCommitsAndWorkItems")),
-                (error) => tl.warning(tl.loc("CommitsAndWorkItemsDownloadFailed", error)));
+            try {
+                let jenkinsClient: JenkinsRestClient = new JenkinsRestClient();
+                let jenkinsJobDetails: JenkinsJobDetails = await jenkinsClient.GetJobDetails();
```

Comment:
should you put this in a function ? its repeated twice.

Reply:
It's only 2 lines, so I guess, we can skip it?

Reply:
COnsider doing it conditionally only for propogated artifacts